functions/src/modules/SQR.functions.ts

```diff
@@ -32,23 +33,35 @@ export const importFilesFromStorage = functions.storage.object()
 
 export const updateFilesOnNewAllotment = functions.database.ref('/sqr/allotments/{allotment_id}')
 .onCreate((snapshot, context) => {
-    const original = snapshot.val();
+    const allotment = snapshot.val();
     let newDocKey = snapshot.key;
-    original.files.forEach(async file => {
-        let file_ref = db.ref(`/sqr/files/${original.list}/${file}`);
+    allotment.files.forEach(async file => {
+        let file_ref = db.ref(`/files/${allotment.list}/${file}/soundQualityReporting`);
         const snapshot = await file_ref.child("status").once('value');
         if(snapshot.exists()) {
-            file_ref.update(
-            {
+            file_ref.update({
                 status: 'Given',
                 allotment: {
                     timestampGiven: new Date().getTime(),
                     timestampDone: null,
-                    devotee: original.devotee
+                    asignee: allotment.asignee
+                }
+            }, (err) => {
+                if (!err) { // if Successful FILE Update, update the ALLOTMENT accordingly
+
+                    // case 1 -- the allotmnet is read from the spreadsheet
+                    if (Object.keys(allotment).indexOf('sendNotificationEmail') > -1)
```

Comment:
I don't understand the condition here.

Reply:
Once a new allotment is added manually I update it to include a `sendNotificationEmail` flag, unlike the ones read from the spreadsheet.

So if the allotment has this `sendNotificationEmail` **property** it means it's manually added and **requires an email notification**.
But if it doesn't, then no emails should be sent as you mentioned in the checklist [here](https://trello.com/c/bbuzNU2l/11-transfer-sqr-data-from-the-spreadsheet-into-database)